### PR TITLE
Keep the correct type of the operand (Fix for issue 207)

### PR DESCRIPTION
**Pull request by clementval**
#207

### CODE REVIEW

File: F-FrontEnd/test/testdata/issue207.f90 [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+
+SUBROUTINE sub1
+IMPLICIT NONE
+  LOGICAL :: log1
+  REAL :: r1
+
+  logext = .TRUE.
+  log1 = (rmod() < 2.0) .OR. logext
+
+END SUBROUTINE sub2
```
Comment (shingo-s) [inline F-FrontEnd/test/testdata/issue207.f90]: sub2 -> sub1

File: F-FrontEnd/test/testdata/issue207.f90 [hunk 2]
```diff
@@ -0,0 +1,24 @@
+MODULE mod1
+
+LOGICAL :: logext
+
+IMPLICIT NONE
```
Comment (shingo-s) [inline F-FrontEnd/test/testdata/issue207.f90]: `IMPLICIT NONE` should appear before all declarations

File: F-FrontEnd/src/F-compile-expr.c [hunk 3]
```diff
@@ -742,9 +742,26 @@ compile_expression(expr x)
...
+                // operand type is NOT FIXED.
                 TYPE_SET_NOT_FIXED(bottom_type(tp));
+                
+                // Re-apply the attribute on the left operand
+                if(bLtype_fixed) {
+                    TYPE_UNSET_NOT_FIXED(bLType);
+                } 
+                // Re-apply the attribute on the right operand
+                if(bRtype_fixed) {
+                    TYPE_UNSET_NOT_FIXED(bRType);
```
Comment (shingo-s) [inline F-FrontEnd/src/F-compile-expr.c]: #207 seems to be occured if `bRType == bottom_type(tp)`.
So this line just reverts `TYPE_SET_NOT_FIXED(bottom_type(tp));` in line 754 and removes the `TYPE_IS_NOT_FIXED` attribute of `bottom_type(tp)`.

Comment (clementval) [reply F-FrontEnd/src/F-compile-expr.c]: @shingo-s So what would be your suggestion?

Comment (shingo-s) [reply F-FrontEnd/src/F-compile-expr.c]: I think `bottom_type(tp)` and `bRType` should be difference objects.
So replacing code between line 731 and line 734 like the following code will solve #207 .

```c
                if((biop == LOGIB || bType == bLType) && shape == lshape) {
                    tp = wrap_type(lt); // This `tp` turns into `bottom_type(tp)` later and `lt` == `bLType` if lt is not an array type
                } else if ((biop == LOGIB || bType == bRType) && shape == rshape) {
                    tp = wrap_type(rt); // same as above
```

Comment (clementval) [reply F-FrontEnd/src/F-compile-expr.c]: @shingo-s Ok, I'll do the change and check it. Thanks for the suggestion.

**Comment by clementval** [comment]
@shingo-s @h-murai Indeed your suggestion works as well.

**Comment by clementval** [comment]
@h-murai Any other change requested or this PR is fine?

**Pull request merged**